src/psyGen.py

```diff
@@ -1819,6 +2051,11 @@ def __init__(self):
         self._fs_descriptors = None
         self._reduction = None
 
+    @property
+    def dag_name(self):
+        ''' Return the name to use in a dag for this node'''
+        return "builtin_" + str(self.abs_position)
```

Comment:
I did this then realized we are using the kernel name for builtins (hence they are all called xx_code) when we should be using the metadata type name i.e. the one we specify in the algorithm layer. I had to change a few tests but all tests are passing now. I guess I could have done this as a different issue but I thought I might as well while I'm here!